moshi/src/main/java/com/squareup/moshi/Moshi.kt

```diff
@@ -0,0 +1,388 @@
+/*
+ * Copyright (C) 2014 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.moshi
+
+import com.squareup.moshi.Types.createJsonQualifierImplementation
+import com.squareup.moshi.internal.Util
+import com.squareup.moshi.internal.Util.canonicalize
+import com.squareup.moshi.internal.Util.isAnnotationPresent
+import com.squareup.moshi.internal.Util.removeSubtypeWildcard
+import com.squareup.moshi.internal.Util.typeAnnotatedWithAnnotations
+import com.squareup.moshi.internal.Util.typesMatch
+import java.lang.reflect.Type
+import javax.annotation.CheckReturnValue
+
+/**
+ * Coordinates binding between JSON values and Java objects.
+ *
+ * Moshi instances are thread-safe, meaning multiple threads can safely use a single instance
+ * concurrently.
+ */
+public class Moshi internal constructor(builder: Builder) {
+  private val factories: List<JsonAdapter.Factory> = buildList {
```

Comment:
Don't recall if `buildList` and `buildSet` are stable already or not, if not - should we avoid using them?

Reply:
they're stable now 👍 

https://kotlinlang.org/docs/whatsnew16.html#stable-collection-builders